### ISSUE TITLE
feature request: temporary location sharing with a public URL?

### ISSUE THREAD

**Issue by kylrth** [bfabiszewski/ulogger-server#190]
I'm interested in uLogger but the one feature I'm missing from a location tracking app is a way to temporarily share location with a public URL, like Facebook Messenger and other apps can. Is this something you'd consider adding?

**Comment by bfabiszewski** [comment]
Yes, I want to have such feature, even for my own use. Just need to find some time to do it

**Comment by europrimus** [comment]
I think about a possible implementation.
An URL like this **https://ulogger.tld/shortlink/bc6d0bcc-6aef-480c-a9e9-c12e470a3d28**
With a table ```short_link``` in database for link to ```tracks.id```

Maybe a *not after* to auto disable short link ?

What about the length of the short link id ? 12 or 24 or 36 char
Does a human going to write it ? if yes, shorter is better.
What the type of char ? number (0-9) and alphabetical (a-z)

**Comment by europrimus** [comment]
hello,
I have started working on it.
you can see what I have done on my [fork](https://github.com/europrimus/ulogger-server/tree/quick_link)

done:
 - setup database
 - read data with GET request `http://ulogger.server/utils/getpositions.php?ql=abc123`

to do:
 - migration for already installed database
 - display positions on the map
 - hide user and track selector and maybe other menu option
 - don't display if after the `not_after` date
 - unit test

if you have some comment, feel free to write.

**Comment by kylrth** [comment]
Hey, this is great! I really appreciate the work you've put in. I agree with the todos as well; ideally most menu options and user info should be hidden.

As I'm a newcomer to the project, I just want to make sure I understand: these links will work *live* as well as *historically*, correct? I can share a link to show old data with a time window that's already finished, and I can also send a link that will show my live location starting from a certain timestamp and ending at a predetermined time?

**Comment by europrimus** [comment]
Quick links work on all tracks individually. But the track need to be already started.
But there is no link to last position or last track. It's not an "I'm here" link.

Use cases examples:

> You go out, but you don't  know witch path you going to take. You start recording your track, share the quick link and every one have the quick link can see witch path you take with your last know position.

> Mister A want to go to this lake, you go there some months or years ago. You share the quick link to your past track with a not after date to tomorrow. And Mister A can walk on your path. But, after tomorrow, he can't see again your track.

I forgot something important on the to do list:
 - create quick link in the android app
 - create quick link in the web site

**Comment by kylrth** [comment]
Ok, I think this is what I expected. Very cool.

**Comment by klauweg** [comment]
That sounds interesting. But as far as i can see it isn't usable already?

**Comment by kylrth** [comment]
Yes, their fork is not quite usable yet.

**Comment by tetricky** [comment]
This, or a similar, feature would be handy. I have a private server, where family are able to see where I am. I would like, on occasion, to be able to share location and gpx tracks with less 'trusted' people. I might only want them to see where I am occasionally.

Would it be possible/easier to be able to synchronise to more than one ulogger server? Then I could run different instances for different access profiles. Toggling on or off synchronisation to different servers depending on when I might want to be seen. Or not.

ulogger is brilliant, and thank you for it. I am using it, and it is invaluable for peace of mind for my partner who suffers from anxiety, but has the reassurance of knowing where I am.

I just don't want other people to know what she knows, all the time... But there are occasions when I would like to share my location with others.

By no means a demand, just wondering if something like that might be an easier implementation. I shall be continuing to use ulogger in its current form irrespective. It's been rock solid and invaluable, and thank you for it.
